src/vivarium/framework/resource/resource.py

```diff
@@ -17,16 +17,33 @@ def resource_id(self) -> str:
         """The long name of the resource, including the type."""
         return f"{self.resource_type}.{self.name}"
 
+    # TODO [MIC_TBD]: make this an abstract method when support for old
+    #  requirements specification is dropped
+    @property
+    def is_initialized(self) -> bool:
+        """Return True if the resource needs to be initialized."""
+        return False
```

Comment:
I guess I don't understand the `is_initialized` logic. How can this always be false for any Resource (besides Null an Columns)?

Reply:
Initializers are always a component's `on_initialize_simulants` method. These are iterated over during simulation initialization (and are the entire reason this resource management system is required). Pipelines and RandomnessStreams don't need to be initialized - they just require certain resources to be present in order for them to be called.